### PR TITLE
Feature/add dectection intelligent video recording

### PR DESCRIPTION
**Pull request by dutsuwak**
* Replaces complex pipeline with inferencebin element
* Adds support for detection on intelligent-video-recording demo
* Adds configuration files for detection or classification

### CODE REVIEW

File: intelligent-video-recording/config_detection.cfg [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+CAMERA_DEVICE = /dev/video0
+CAMERA_DEVICE_WIDTH = 1280
+CAMERA_DEVICE_HEIGHT = 720
+ARCH=mobilenetv2ssd
+MODEL_LOCATION = /home/coral/models/ssd_mobilenet_v2_coco_quant_postprocess_edgetpu.tflite
+INPUT_LAYER = 'input'
+OUTPUT_LAYER = 'MobilenetV2/Predictions/Reshape_1'
+LABELS = coco_labels.txt
+OUTPUT_FILE = /home/coral/recordings/recording.mp4
+CLASSES_ID = [74]
```
Comment (rrcarlosrodriguez) [inline intelligent-video-recording/config_detection.cfg:17]: I think we should explain better what these numbers are in the README file. We should assume that people don't know that a number corresponds to a label and corresponds to the kind of object we will be tracking

File: intelligent-video-recording/gst_display.py [hunk 2]
```diff
@@ -106,29 +105,51 @@ def parseLabels(self, file):
     def newPrediction(self, element, meta):
         data = json.loads(meta)
 
-        # Parse class id from prediction
```
Comment (rrcarlosrodriguez) [inline intelligent-video-recording/gst_display.py:109]: Document this newPrediction function. Put a comment explaining how is it being used, what is generated by the inferencebin element and when. Remember this demo is intended to show how GstInference works for someone who has never heard about it

**Review by rrcarlosrodriguez — CHANGES_REQUESTED** [review]

File: intelligent-video-recording/README.md [hunk 3]
```diff
@@ -13,11 +13,13 @@ The demo uses RidgeRun open source project GstInterpipes. Please make sure you i
 
 * [GStreamer Interpipes](https://developer.ridgerun.com/wiki/index.php?title=GstInterpipe_-_Building_and_Installation_Guide)
 
-The demo requires the model and labels file from Coral which can be downloaded from:
 
-[https://coral.ai/models/](https://coral.ai/models/)
+Update the paths to model and labels file on ``config_<type>.cfg`` file.
```
Comment (aphillips98) [inline intelligent-video-recording/README.md]: Check: Update the paths for the model and label files...

File: intelligent-video-recording/README.md [hunk 4]
```diff
@@ -13,11 +13,13 @@ The demo uses RidgeRun open source project GstInterpipes. Please make sure you i
...
+A configuration file is available for each demo <type> (classification, detection):
 
-Update the paths to model and labels file on ``config.cfg`` file. It uses by default ``mobilenet_v2_1.0_224_quant_edgetpu.tflite`` and ``imagenet_labels.txt``.
+* config_detection.cfg: Uses MobileNetV2 + SSD (detection) files available here: [labels](https://developer.ridgerun.com/wiki/index.php?title=Coral_MobilenetV2SSD_COCO_labels) and [models](https://coral.ai/models/).
+In this case, you need to save the labels content into a file named ``coco_labels.txt``.
```
Comment (aphillips98) [inline intelligent-video-recording/README.md]: Check: label contents into a file called...

**Review by bill9889 — CHANGES_REQUESTED** [review]

File: intelligent-video-recording/gst_display.py [hunk 5]
```diff
@@ -46,27 +46,32 @@ def __init__(self, config_file_name):
...
+            self.arch = self.config['DEMO_SETTINGS']['ARCH']
             self.classes_id = ast.literal_eval(
                                 self.config['DEMO_SETTINGS']['CLASSES_ID'])
             self.classes_probability = ast.literal_eval(
                                 self.config['DEMO_SETTINGS']['CLASSES_MIN_PROBABILITY'])
             self.min_recording_time_seconds = int(self.config['DEMO_SETTINGS']\
                                 ['MIN_RECORDING_TIME_IN_SECONDS'])
+
+            if(len(self.classes_id) != len(self.classes_probability)):
+                print(("Classes_ID and Class_Min_Probability list must be of the same"
```
Comment (michaelgruner) [inline intelligent-video-recording/gst_display.py:59]: Print all errors to stdout.

File: intelligent-video-recording/gst_display.py [hunk 6]
```diff
@@ -46,27 +46,32 @@ def __init__(self, config_file_name):
...
+                       " lenght. Please review configuration file."))
+                exit(0)
```
Comment (michaelgruner) [inline intelligent-video-recording/gst_display.py]: Return a code other than 0 in the case of an error.

File: intelligent-video-recording/config_detection.cfg [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# Use absolute path for model location and output file
+# Only MP4 recording is supported by the demo
+[DEMO_SETTINGS]
+CAMERA_DEVICE = /dev/video0
+CAMERA_DEVICE_WIDTH = 1280
+CAMERA_DEVICE_HEIGHT = 720
+ARCH=mobilenetv2ssd
+MODEL_LOCATION = /home/coral/models/ssd_mobilenet_v2_coco_quant_postprocess_edgetpu.tflite
+INPUT_LAYER = 'input'
+OUTPUT_LAYER = 'MobilenetV2/Predictions/Reshape_1'
```
Comment (michaelgruner) [inline intelligent-video-recording/config_detection.cfg]: Please double check if the input and output names are needed in the coral. If not, please leave blank and leave a comment saying this is not needed for the coral, but other backends (such as TensorFlow) may need it.

File: intelligent-video-recording/gst_display.py [hunk 8]
```diff
@@ -104,31 +109,58 @@ def parseLabels(self, file):
...
+                    else:
+                        self.start_recording_time = time.time()
+            else:
+                if(self.recording):
+                    self.stop_recording_time = time.time()
+                    diff = self.stop_recording_time - self.start_recording_time
+
+                    if(diff >= self.min_recording_time_seconds):
+                        print("No detection. Stop recording.")
+                        self.stopRecordingPipeline()
```
Comment (michaelgruner) [inline intelligent-video-recording/gst_display.py]: I believe this logic is flawed. The recording will only stop if there is a new prediction. But in the case were a person walks out of the scene and no new predictions are made, the recording will continue forever. We need a timeout mechanism that is restarted with every new valid detection.

**Review by michaelgruner — CHANGES_REQUESTED** [review]

File: intelligent-video-recording/gst_display.py (hunk 5 above)
Comment (michaelgruner) [reply intelligent-video-recording/gst_display.py:59]: sorry, I meant stderr

**Review by michaelgruner — APPROVED** [review]

**Pull request merged**